File: app/backend/src/main/kotlin/at/htlleonding/maturaballmanager/model/dtos/InvoiceDTO.kt
import at.htlleonding.maturaballmanager.model.Status
import java.time.OffsetDateTime

data class InvoiceDTO(
    var id: String? = null,
    var company: String? = null,
    var contactPerson: String? = null,

    var companyName: String? = null,
    var contactPersonName: String? = null,

    var benefits: List<String> = emptyList(),
    var invoiceDate: OffsetDateTime? = null,
    var paymentDeadline: OffsetDateTime? = null,
    var status: Status = Status.DRAFT,
    var totalAmount: Double? = 0.0,
    var sendOption: String? = "immediate"
)

File: app/frontend/web/src/app/model/dtos/invoice.dto.ts
import {Invoice, Status} from "../invoice";

export interface InvoiceDTO {
    id?: string;
    company?: string;
    contactPerson?: string;
    benefits?: string[];
    invoiceDate?: Date;
    paymentDeadline?: Date;
    status?: Status;
    totalAmount?: number;
    sendOption?: 'immediate' | 'onDate';
}
